Document paper endpoints (#1310)
huggingface/hub-docs: docs/hub/api.md

This is equivalent to `huggingface_hub.whoami()`.

## Paper Pages API

The following endpoint gets information about a paper.

### GET /api/arxiv/{arxiv_id}/repos

> Get all the models, datasets, and Spaces mentioned in a paper.

Reviewer: we won't be able to use the hits for this to represent actual calls from Arxiv 

but i guess that's fine?

Author: From my side it's fine

Reviewer: BTW this is not really what this is doing

```suggestion
### GET /api/arxiv/{arxiv_id}/repos

Get all the models, datasets, and Spaces that refer to a paper.
```

better?^
